src/Microsoft.Framework.FileSystemGlobbing/Internal/PatternContexts/PatternContextRagged.cs

```diff
@@ -81,19 +70,49 @@ public override void PushFrame(DirectoryInfoBase directory)
                 }
             }
 
-            PushFrame(frame);
+            PushDataFrame(frame);
+        }
+
+        public struct FrameData
+        {
+            public bool IsNotApplicable;
+
+            public int SegmentGroupIndex;
+
+            public IList<IPathSegment> SegmentGroup;
+
+            public int BacktrackAvailable;
+
+            public int SegmentIndex;
+        }
+
+        protected IRaggedPattern Pattern { get; }
+
+        protected bool IsStartsWith()
```

Comment:
I've seen that you converted some basic Getter properties to methods.  Do methods perform better than properties? ( I don't know much about the internals of properties)